vm/state_manager/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#![allow(dead_code)]
+
+mod errors;
+
+pub use self::errors::*;
+use actor::{MinerInfo, StorageMinerActorState};
+use address::Address;
+use amt::BlockStore;
+use chain::ChainStore;
+use encoding::de::DeserializeOwned;
+use state_tree::{HamtStateTree, StateTree};
+
+/// Intermediary for retrieving state objects and updating actor states
+pub struct StateManager<'a> {
+    cs: &'a ChainStore<'a>,
```

Comment:
My worry is that the chainstore will have to be mutable (updating heaviest tipset) and requiring an immutable reference for this type will inhibit this, thoughts?

Reply:
Yeah, I agree with your concern, will look to improve.